jupyterhub_moss/spawner.py

```diff
@@ -321,12 +321,29 @@ def __update_options(
                 raise RuntimeError("GPU(s) not available for this partition")
             options.gres = gpu_gres_template.format(options.ngpus)
 
-        # Custom envs are always added to PATH
-        prologue_env_path = options.environment_path
-        # Default envs only added to PATH if add_to_path is True
-        partition_default_envs = partition_info.jupyter_environments
-        if options.environment_id in partition_default_envs:
-            if not partition_default_envs[options.environment_id].add_to_path:
+        # Use first env from the partition if none is requested
+        if (
+            not options.environment_id
+            and not options.environment_path
+            and not options.environment_modules
+        ):
+            default_env = list(partition_info.jupyter_environments.keys())[0]
+            options.environment_id = default_env
```

Comment:
This restores usage of the "first" env of the partition as a default. That was lost in PR #99.